prymer/api/oligo.py

```diff
@@ -95,6 +112,11 @@ class Oligo(OligoLike, Metric["Oligo"]):
         bases: the base sequence of the oligo (excluding any tail)
         tail: an optional tail sequence to put on the 5' end of the primer
         name: an optional name to use for the primer
+        self_any_th: an optional calculated melting temperature that represents
+            the tendency of an oligo to bind to itself (self-complementarity)
+        self_end_th: an optional calculated melting temperature that represents
+            the tendency of a primer to bind to the 3'-END of an identical primer
+        hairpin_th: an optional calculated melting temperature of the oligo hairpin structure
```

Comment:
What do others think about giving these names that are more descriptive vs. sticking with what primer3 calls them?  I would propose:

- `self_any_th` -> `homodimer_tm`
- `self_end_th` -> `3p_anchored_homodimer_tm`
- `hairpin_th` -> `secondary_structure_tm`

Reply:
For me, I think this comes down to ... is this library solely about wrapping primer3, or more generally for primer design.  But I agree, that if we rename, we should also reference the primer3 return value/key/concept that matches this in the documentation.

Reply:
Is there another primer design tool you've used with `fgprimer` in the past?

Reply:
decided:
* self_any_th -> tm_homodimer
* self_end_th -> tm_3p_anchored_homodimer
* hairpin_th -> tm_secondary_structure